generate-xml/domain.nix

```diff
@@ -218,6 +218,33 @@ let
                   targetelem
                   addresselem
                 ])
+              (subelem "hostdev"
+                [
+                  (subattr "mode" typeString)
+                  (subattr "type" typeString)
+                  (subattr "managed" typeBoolYesNo)
+                ]
+                [
+                  (subelem "source" [] [ 
+                    # Could not be "addresselem" because it uses hex values like "0x0000"
```

Comment:
The libvirt domain XML allows plain integers in `<address>` elements, not just hex values. So you should be able to use `addresselem`.

Reply:
I mean it NEEDS hex values and NOT integer? Does it support hex?

Reply:
As far as I know, all the XML `<address>` element attributes that accept hex values also accept plain integers.

The `typeInt` function accepts an integer, but not a string. If it's OK to pass in an integer there, you can use `addresselem`. Just put `0` there instead of `"0x0000"`.

Reply:
I understand. I try it and then change it.